pkg/validation/validate.go

```diff
@@ -0,0 +1,31 @@
+package validation
+
+import "reflect"
+
+func Validate(s any, isNew bool) ErrorList {
```

Comment:
I don't really like that objects have a `Validate` method. In my opinion this violates the **S**OLID (Single-Responsibility) principle. Validation logic should be in its own encapsulated function/component/class. That will become increasingly evident once validation logic needs to make database calls, and then we need to inject a database handle into each Data Transfer Object (dto).